include/deal.II/lac/la_vector.templates.h

```diff
@@ -398,7 +398,7 @@ namespace LinearAlgebra
 
     // in >> c;
     in.read(&c,1);
-    AssertThrow(c=']', ExcIO());
+    AssertThrow(c==']', ExcIO());
```

Comment:
Not weird at all. Code gets copied around. I asked because I expected something like what you found.